### DISCUSSION TITLE
How is frequency calculated in topics per class?

### DISCUSSION THREAD

**Discussion by chrislalk** [MaartenGr/BERTopic#854] [Q&A]
Hi all! 

I wonder how frequency is calculated in topics per class and how does it relate to the probabilities? Intuitively, I would have guessed that I can aggregate the probabilities of a topic over all documents in a class and use this as my expected value for this class similar to frequency. However, then I wouldn't necessarily get integers as in frequency. Not sure though if that makes sense. 
Any thoughts?

**Comment by MaartenGr** [comment]
> I wonder how frequency is calculated in topics per class and how does it relate to the probabilities?

The frequencies in the `.topics_per_class` function is simply the count of documents in that specific class across each topic. In other words, it is the number of documents for topic *t* in class *c*. Thus, this frequency has no relation at all with the probabilities. 

> Intuitively, I would have guessed that I can aggregate the probabilities of a topic over all documents in a class and use this as my expected value for this class similar to frequency.

Could you elaborate a bit more on this? I am not sure if I understood this correctly. Do you want the probability of a topic belonging to a specific class?
